core/common/src/main/java/alluxio/security/authentication/plain/PlainSaslServerCallbackHandler.java

```diff
@@ -84,6 +84,8 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
 
       // After verification succeeds, a user with this authz id will be set to a Threadlocal.
       AuthenticatedClientUser.set(ac.getAuthorizedID());
+      // The user who is connecting to the alluxio's server will be set to a Threadlocal too.
+      AuthenticatedClientUser.setConnectionUser(username);
```

Comment:
I'll be honest, I looked into this but I'm very in the dark about how changing `AuthenticatedClientUser.sConnectionUserThreadLocal` may impact other parts of the system so I can't really approve nor deny these changes.

Reply:
The code is in the function `alluxio.security.authentication.AuthenticatedUserInjector#authenticateCall`:
```java
          AuthenticatedClientUser.set(userInfo.getAuthorizedUserName());
          AuthenticatedClientUser.setConnectionUser(userInfo.getConnectionUserName());
          AuthenticatedClientUser.setAuthMethod(userInfo.getAuthMethod());
```
so, if the permission is passed, we should set the connection user in function `alluxio.security.authentication.plain.SaslServerHandlerPlain#getAuthenticatedUserInfo`:
```java
  @Override
  public AuthenticatedUserInfo getAuthenticatedUserInfo() {
    if (SecurityUtils.isAuthenticationEnabled(ServerConfiguration.global())) {
      try {
        return new AuthenticatedUserInfo(mSaslServer.getAuthorizationID(),
                AuthenticatedClientUser.getConnectionUser(ServerConfiguration.global()));
      } catch (AccessControlException e) {
        // do nothing
      }
    }
    return new AuthenticatedUserInfo(mSaslServer.getAuthorizationID());
  }
```